metrics/metric.go

```diff
@@ -0,0 +1,87 @@
+package metrics
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type Metric struct {
+	name       string
+	labels     []string
+	labelCount int
+
+	generators map[string]ValueGenerator
+
+	sample       *prometheus.GaugeVec
+	promRegistry *prometheus.Registry
+
+	cycleId int
+}
+
+func NewMetrics(registery *prometheus.Registry, name string, labelCount int, defaultCardinality int, cardinalityMap map[string]int) Metric {
+	var labels []string
+	for idx := 0; idx < labelCount; idx++ {
+		label := "label_key_" + strconv.Itoa(idx)
+		labels = append(labels, label)
+	}
+	var metric Metric
+	log.Printf("Creating new metric with name %s and %d labels", name, labelCount)
+	metric.labelCount = labelCount
```

Comment:
Do you need to store labelCount here? won't array length be sufficient?

Reply:
Removed